### ISSUE TITLE
Disabling logging via isDebugEnabled=false does not seem to work

### ISSUE THREAD

**Issue by sowens-csd** [ConnectyCube/connectycube-flutter-sdk-releases#39]
**Platform (use [x])**
[ ] Android
[ x] iOS

**Device info**
Manufacture: Apple
Model: iPad Air (3rd gen)
OS version: 14.4

**Describe the bug:**
This may not be a bug, perhaps I'm just misunderstanding the documentation. I'd like to disable most or all of the ConnectyCube logging unless I'm trying to debug a problem. The quantity of logs makes it hard to understand the rest of the flow of my app. 

Despite setting 
```dart
    CubeSettings.instance.isDebugEnabled = false;
```
I still see detailed logs from ConnectyCube in the console. 

**Steps to Reproduce:**
1. Checkout the ConnectCube sample apps
2. In p2p_call_sample make a change in main.dart to set the debugging state before the `init` call.
3. Run the app 
4. Choose one of the logins

**Logs:**
```
Launching lib/main.dart on LuckySPOPad (2) in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: 3X949YE9K2
Xcode build done.                                           26.3s
Connecting to VM Service at ws://localhost:55348/ws
[38;5;244m[log] [39;49mD/[Connection]: State: XmppConnectionState.SocketOpening
[38;5;244m[log] [39;49mD/[Connection]: State: XmppConnectionState.SocketOpened
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<?xml version='1.0'?>
      <stream:stream xmlns='jabber:client' version='1.0' xmlns:stream='http://etherx.jabber.org/streams'
      to='chat.connectycube.com'
      xml:lang='en'
      >
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<?xml version='1.0'?><stream:stream xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' from='chat.connectycube.com' id='03dadc65-9a58-4f5d-9881-2f42efce86bf' version='1.0' xml:lang='en'>
[38;5;244m[log] [39;49mD/[Connection]: processInitialStream
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<stream:features><sm xmlns="urn:xmpp:sm:3"/><mechanisms xmlns="urn:ietf:params:xml:ns:xmpp-sasl"><mechanism>PLAIN</mechanism><mechanism>ANONYMOUS</mechanism><mechanism>PLAIN_FAST</mechanism></mechanisms><ver xmlns="urn:xmpp:features:rosterver"/><starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls"/></stream:features>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Negotiating features
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator true
[38;5;244m[log] [39;49mD/[StartTlsNegotiator]: negotiating starttls
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<starttls xmlns="urn:ietf:params:xml:ns:xmpp-tls"/>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: ACTIVE FEATURE: {name: StartTlsNegotiator, name_space: urn:ietf:params:xml:ns:xmpp-tls, priority: 1, state: NegotiatorState.NEGOTIATING}, isReady: true
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<proceed xmlns="urn:ietf:params:xml:ns:xmpp-tls"/>
[38;5;244m[log] [39;49mD/[Connection]: startSecureSocket
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<?xml version='1.0'?>
      <stream:stream xmlns='jabber:client' version='1.0' xmlns:stream='http://etherx.jabber.org/streams'
      to='chat.connectycube.com'
      xml:lang='en'
      >
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<?xml version='1.0'?><stream:stream xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' from='chat.connectycube.com' id='03dadc65-9a58-4f5d-9881-2f42efce86bf' version='1.0' xml:lang='en'>
[38;5;244m[log] [39;49mD/[Connection]: processInitialStream
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<stream:features><sm xmlns="urn:xmpp:sm:3"/><mechanisms xmlns="urn:ietf:params:xml:ns:xmpp-sasl"><mechanism>PLAIN</mechanism><mechanism>ANONYMOUS</mechanism><mechanism>PLAIN_FAST</mechanism></mechanisms><ver xmlns="urn:xmpp:features:rosterver"/></stream:features>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Negotiating features
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator true
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<auth xmlns="urn:ietf:params:xml:ns:xmpp-sasl" mechanism="PLAIN">ADEyNTMxNTgtNDc2AHh4YXNCVU0zZ1FzMzZiaGo=</auth>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: ACTIVE FEATURE: {name: SaslAuthenticationFeature, name_space: null, priority: 1000, state: NegotiatorState.NEGOTIATING}, isReady: true
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<success xmlns="urn:ietf:params:xml:ns:xmpp-sasl"/>
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<?xml version='1.0'?>
      <stream:stream xmlns='jabber:client' version='1.0' xmlns:stream='http://etherx.jabber.org/streams'
      to='chat.connectycube.com'
      xml:lang='en'
      >
[38;5;244m[log] [39;49mD/[Connection]: State: XmppConnectionState.Authenticated
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator false
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: No matching negotiator
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<?xml version='1.0'?><stream:stream xmlns='jabber:client' xmlns:stream='http://etherx.jabber.org/streams' from='chat.connectycube.com' id='03dadc65-9a58-4f5d-9881-2f42efce86bf' version='1.0' xml:lang='en'>
[38;5;244m[log] [39;49mD/[Connection]: processInitialStream
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<stream:features><sm xmlns="urn:xmpp:sm:3"/><mobile xmlns="http://tigase.org/protocol/mobile#v2"/><ver xmlns="urn:xmpp:features:rosterver"/><bind xmlns="urn:ietf:params:xml:ns:xmpp-bind"/><session xmlns="urn:ietf:params:xml:ns:xmpp-session"/></stream:features>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Negotiating features
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator false
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator true
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<iq id="GHPKCXBHE" type="set">
        <bind xmlns="urn:ietf:params:xml:ns:xmpp-bind">
          <resource/>
        </bind>
      </iq>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: ACTIVE FEATURE: {name: BindingResourceConnectionNegotiator, name_space: null, priority: 100, state: NegotiatorState.NEGOTIATING}, isReady: true
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<iq xmlns="jabber:client" id="GHPKCXBHE" to="1253158-476@chat.connectycube.com/1204607085-chat-34113" type="result"><bind xmlns="urn:ietf:params:xml:ns:xmpp-bind"><jid>1253158-476@chat.connectycube.com/1204607085-chat-34113</jid></bind></iq>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator true
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<enable xmlns="urn:xmpp:sm:3" resume="true"/>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: ACTIVE FEATURE: {name: StreamManagementModule, name_space: null, priority: 1000, state: NegotiatorState.NEGOTIATING}, isReady: true
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Feature Started Parsing
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<enabled xmlns='urn:xmpp:sm:3' id='b853ffa3-c47f-4123-9548-96ff5ed6b25c' resume='true' max='60' location='localhost' />
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator false
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator true
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<iq id="DHWFDSKOA" type="set" to="chat.connectycube.com">
        <session xmlns="urn:ietf:params:xml:ns:xmpp-session"/>
      </iq>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: ACTIVE FEATURE: {name: SessionInitiationNegotiator, name_space: null, priority: 1000, state: NegotiatorState.NEGOTIATING}, isReady: true
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<iq xmlns="jabber:client" id="DHWFDSKOA" to="1253158-476@chat.connectycube.com/1204607085-chat-34113" type="result" from="chat.connectycube.com"/>
[38;5;244m[log] [39;49mD/[Connection]: State: XmppConnectionState.SessionInitialized
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator false
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator true
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<iq id="CEPNRNOST" type="get" from="1253158-476@chat.connectycube.com/1204607085-chat-34113" to="chat.connectycube.com">
        <query xmlns="http://jabber.org/protocol/disco#info"/>
      </iq>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: ACTIVE FEATURE: {name: ServiceDiscoveryNegotiator, name_space: null, priority: 1000, state: NegotiatorState.NEGOTIATING}, isReady: true
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="1"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<iq xmlns="jabber:client" id="CEPNRNOST" to="1253158-476@chat.connectycube.com/1204607085-chat-34113" type="result" from="chat.connectycube.com"><query xmlns="http://jabber.org/protocol/disco#info"><identity name="App Chat ver. 2.1.0-0" category="component" type="im"/><identity name="App Chat ver. 2.1.0-0" category="server" type="im"/><feature var="http://jabber.org/protocol/commands"/><feature var="urn:xmpp:carbons:2"/><feature var="http://jabber.org/protocol/stats"/><feature var="http://jabber.org/protocol/amp"/><feature var="msgoffline"/><feature var="http://jabber.org/protocol/disco#info"/><feature var="http://jabber.org/protocol/disco#items"/><feature var="urn:xmpp:ping"/><feature var="urn:ietf:params:xml:ns:xmpp-sasl"/><feature var="jabber:iq:roster"/><feature var="jabber:iq:roster-dynamic"/><feature var="jabber:iq:privacy"/><feature var="urn:ietf:params:xml:ns:xmpp-bind"/><feature var="http://jabber.org/protocol/commands"/><feature var="jabber:iq:last"/><feature var="urn:ietf:params:xml:ns:xmpp-session"/></query></iq>
[38;5;244m[log] [39;49mE/[ConnectionNegotiatorManager]: ADDING FEATURES count: 16 [Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature', Instance of 'Feature']
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Adding negotiator: {name: urn:xmpp:carbons, name_space: null, priority: 1000, state: NegotiatorState.IDLE}, isReady: true [Instance of 'Feature']
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator false
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator true
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<iq id="ELLNFARIF" type="set" xmlns="jabber:client">
        <enable xmlns="urn:xmpp:carbons:2"/>
      </iq>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: ACTIVE FEATURE: {name: urn:xmpp:carbons, name_space: null, priority: 1000, state: NegotiatorState.NEGOTIATING}, isReady: true
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="2"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<iq xmlns="jabber:client" id="ELLNFARIF" to="1253158-476@chat.connectycube.com/1204607085-chat-34113" type="result"/>
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: Found matching negotiator false
[38;5;244m[log] [39;49mD/[ConnectionNegotiatorManager]: No matching negotiator
[38;5;244m[log] [39;49mD/[Connection]: State: XmppConnectionState.Ready
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<iq id="XMYRDYFPH" type="get">
        <query xmlns="jabber:iq:roster"/>
      </iq>
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<presence/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="3"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<iq xmlns="jabber:client" id="XMYRDYFPH" to="1253158-476@chat.connectycube.com/1204607085-chat-34113" type="result"><query xmlns="jabber:iq:roster"><item name="laironrq" jid="laironrq@conversations.im" subscription="none"/><item name="xshmelkin" jid="xshmelkin@cooljabber.pw" subscription="none"/><item name="relaxf" jid="relaxf@daemons.it" subscription="none"/><item name="xumukxn" jid="xumukxn@conversations.im" subscription="none"/><item name="akutorlx" jid="akutorlx@jabber.fr" subscription="none"/><item name="vpudding" jid="vpudding@conversations.im" subscription="none"/></query></iq>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="4"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<presence xmlns="jabber:client" to="1253158-476@chat.connectycube.com" type="subscribe" from="laironrq@conversations.im"/>
[38;5;244m[log] [39;49mD/[StanzaParser]: No id found for stanza
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="5"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<presence xmlns="jabber:client" to="1253158-476@chat.connectycube.com" type="subscribe" from="xshmelkin@cooljabber.pw"/>
[38;5;244m[log] [39;49mD/[StanzaParser]: No id found for stanza
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="6"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<presence xmlns="jabber:client" to="1253158-476@chat.connectycube.com" type="subscribe" from="relaxf@daemons.it"/>
[38;5;244m[log] [39;49mD/[StanzaParser]: No id found for stanza
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="7"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<presence xmlns="jabber:client" to="1253158-476@chat.connectycube.com" type="subscribe" from="xumukxn@conversations.im"/>
[38;5;244m[log] [39;49mD/[StanzaParser]: No id found for stanza
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="8"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<presence xmlns="jabber:client" to="1253158-476@chat.connectycube.com" type="subscribe" from="akutorlx@jabber.fr"/>
[38;5;244m[log] [39;49mD/[StanzaParser]: No id found for stanza
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="9"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<presence xmlns="jabber:client" to="1253158-476@chat.connectycube.com" type="subscribe" from="vpudding@conversations.im"/>
[38;5;244m[log] [39;49mD/[StanzaParser]: No id found for stanza
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="10"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<presence xmlns="jabber:client" to="1253158-476@chat.connectycube.com" from="1253158-476@chat.connectycube.com/1204607085-chat-34113"/>
[38;5;244m[log] [39;49mD/[StanzaParser]: No id found for stanza
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<r xmlns='urn:xmpp:sm:3' />
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<a xmlns="urn:xmpp:sm:3" h="11"/>
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<r xmlns="urn:xmpp:sm:3"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<a xmlns='urn:xmpp:sm:3' h='5'/>
[38;5;244m[log] [39;49mD/[StreamManagementModule]: Delivered: DHWFDSKOA
[38;5;244m[log] [39;49mD/[StreamManagementModule]: Delivered: CEPNRNOST
[38;5;244m[log] [39;49mD/[StreamManagementModule]: Delivered: ELLNFARIF
[38;5;244m[log] [39;49mD/[StreamManagementModule]: Delivered: XMYRDYFPH
[38;5;244m[log] [39;49mD/[StreamManagementModule]: Delivered stanza without id presence
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<r xmlns="urn:xmpp:sm:3"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<a xmlns='urn:xmpp:sm:3' h='5'/>
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<r xmlns="urn:xmpp:sm:3"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<a xmlns='urn:xmpp:sm:3' h='5'/>
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<r xmlns="urn:xmpp:sm:3"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<a xmlns='urn:xmpp:sm:3' h='5'/>
[38;5;244m[log] [39;49m---Xmpp Sending:---
[38;5;244m[log] [39;49m<r xmlns="urn:xmpp:sm:3"/>
[38;5;244m[log] [39;49m---Xmpp Receiving:---
[38;5;244m[log] [39;49m<a xmlns='urn:xmpp:sm:3' h='5'/>
```
or
[Gist log](https://gist.github.com/)

**Actual result:**
Lots of logging from ConnectyCube

**Expected behavior:**
No logging, or greatly reduced logging, from ConnectyCube

**Additional info**
Here's the changed method:

```dart
@override
  void initState() {
    super.initState();
    CubeSettings.instance.isDebugEnabled = false;
    init(
      config.APP_ID,
      config.AUTH_KEY,
      config.AUTH_SECRET,
    );
  }
```

**Comment by TatankaConCube** [comment]
@sowens-csd thank you for reporting this issue. These logs come from one of the dependency libraries, we will review, how can we disable it from our SDK.

**Comment by TatankaConCube** [comment]
implemented since version [2.0.0-nullsafety.4](https://pub.dev/packages/connectycube_sdk/versions/2.0.0-nullsafety.4)
